### PR TITLE
feat: implement basic parallelization in the one shot workflows

### PR DESCRIPTION
**Pull request by nathanklick**
## Description

This pull request introduces support for a "one-shot" execution mode across several command classes, allowing certain operations (such as lease acquisition and release) to be conditionally skipped when this mode is active. The changes are primarily focused on dependency injection and conditional logic to check the state of `OneShotState`, ensuring that resources like locks are only managed when appropriate. This improves efficiency and correctness in concurrent or stateless command executions.

The most important changes are:

**One-shot mode integration (core logic):**
* Added `OneShotState` as an injectable dependency to `BaseCommand`, `BlockNodeCommand`, `ClusterCommandTasks`, and `ExplorerCommand`, enabling commands to check if one-shot mode is active. [[1]](diffhunk://#diff-d62b22b357d8189368367ee647935585998107d28bfaf657bac900e1087a9597R32) [[2]](diffhunk://#diff-d62b22b357d8189368367ee647935585998107d28bfaf657bac900e1087a9597R53) [[3]](diffhunk://#diff-d62b22b357d8189368367ee647935585998107d28bfaf657bac900e1087a9597R67) [[4]](diffhunk://#diff-bc0352c333cbf78bb4e2e2f07fca1e6370741af44ebc3454b5c6a09ada171e6aR31) [[5]](diffhunk://#diff-bc0352c333cbf78bb4e2e2f07fca1e6370741af44ebc3454b5c6a09ada171e6aR46) [[6]](diffhunk://#diff-bc0352c333cbf78bb4e2e2f07fca1e6370741af44ebc3454b5c6a09ada171e6aR55)
* Updated lease acquisition and release logic in `BlockNodeCommand`, `ClusterCommandTasks`, and `ExplorerCommand` to skip these operations when one-shot mode is active, using `ListrLock.newSkippedLockTask` instead of acquiring a lock. [[1]](diffhunk://#diff-4faedeee6f9618aff5f5c4a02f13c1617c27edc8b67e396d4b2b303e3fb1fa98R395-R397) [[2]](diffhunk://#diff-4faedeee6f9618aff5f5c4a02f13c1617c27edc8b67e396d4b2b303e3fb1fa98R464-R467) [[3]](diffhunk://#diff-4faedeee6f9618aff5f5c4a02f13c1617c27edc8b67e396d4b2b303e3fb1fa98R591-R599) [[4]](diffhunk://#diff-4faedeee6f9618aff5f5c4a02f13c1617c27edc8b67e396d4b2b303e3fb1fa98R616-R618) [[5]](diffhunk://#diff-4faedeee6f9618aff5f5c4a02f13c1617c27edc8b67e396d4b2b303e3fb1fa98R655-R658) [[6]](diffhunk://#diff-4faedeee6f9618aff5f5c4a02f13c1617c27edc8b67e396d4b2b303e3fb1fa98R693-R701) [[7]](diffhunk://#diff-4faedeee6f9618aff5f5c4a02f13c1617c27edc8b67e396d4b2b303e3fb1fa98R718-R720) [[8]](diffhunk://#diff-4faedeee6f9618aff5f5c4a02f13c1617c27edc8b67e396d4b2b303e3fb1fa98R759-R762) [[9]](diffhunk://#diff-4faedeee6f9618aff5f5c4a02f13c1617c27edc8b67e396d4b2b303e3fb1fa98R823-R831) [[10]](diffhunk://#diff-4faedeee6f9618aff5f5c4a02f13c1617c27edc8b67e396d4b2b303e3fb1fa98R848-R850) [[11]](diffhunk://#diff-4faedeee6f9618aff5f5c4a02f13c1617c27edc8b67e396d4b2b303e3fb1fa98R890-R893) [[12]](diffhunk://#diff-4faedeee6f9618aff5f5c4a02f13c1617c27edc8b67e396d4b2b303e3fb1fa98R910-R918) [[13]](diffhunk://#diff-4faedeee6f9618aff5f5c4a02f13c1617c27edc8b67e396d4b2b303e3fb1fa98R935-R937) [[14]](diffhunk://#diff-4faedeee6f9618aff5f5c4a02f13c1617c27edc8b67e396d4b2b303e3fb1fa98R964-R967) [[15]](diffhunk://#diff-4faedeee6f9618aff5f5c4a02f13c1617c27edc8b67e396d4b2b303e3fb1fa98R984-R992) [[16]](diffhunk://#diff-bc0352c333cbf78bb4e2e2f07fca1e6370741af44ebc3454b5c6a09ada171e6aL425-R434) [[17]](diffhunk://#diff-1e98e31967fc6e63e1e65eb39a2b1c00e5d007a95067861acbde89787f995098L568-R574) [[18]](diffhunk://#diff-1e98e31967fc6e63e1e65eb39a2b1c00e5d007a95067861acbde89787f995098R629-R632) [[19]](diffhunk://#diff-1e98e31967fc6e63e1e65eb39a2b1c00e5d007a95067861acbde89787f995098R663-R671) [[20]](diffhunk://#diff-1e98e31967fc6e63e1e65eb39a2b1c00e5d007a95067861acbde89787f995098L669-R690) [[21]](diffhunk://#diff-1e98e31967fc6e63e1e65eb39a2b1c00e5d007a95067861acbde89787f995098R736-R739) [[22]](diffhunk://#diff-1e98e31967fc6e63e1e65eb39a2b1c00e5d007a95067861acbde89787f995098R762-R770)

**Concurrency and configuration handling:**
* In `ExplorerCommand`, added logic to override potentially stale configuration values from `argv` when running in one-shot mode, ensuring correct values are used in concurrent executions.

**Dependency injection and imports:**
* Updated imports and constructor signatures to include `OneShotState` and `Listr` where necessary for the new logic. [[1]](diffhunk://#diff-bc0352c333cbf78bb4e2e2f07fca1e6370741af44ebc3454b5c6a09ada171e6aR3) [[2]](diffhunk://#diff-1e98e31967fc6e63e1e65eb39a2b1c00e5d007a95067861acbde89787f995098R3)

These changes collectively enable commands to efficiently support both regular and stateless (one-shot) execution scenarios, improving robustness in concurrent environments.

### Related Issues

* Related to #3423 

### Pull request (PR) checklist

* \[ ] This PR added tests (unit, integration, and/or end-to-end)
* \[ ] This PR updated documentation
* \[x] This PR added no TODOs or commented out code
* \[x] This PR has no breaking changes
* \[x] Any technical debt has been documented as a separate issue and linked to this PR
* \[x] Any `package.json` changes have been explained to and approved by a repository manager
* \[x] All related issues have been linked to this PR
* \[x] All changes in this PR are included in the description
* \[x] When this PR merges the commits will be squashed and the title will be used as the commit message, the 'commit message guidelines' below have been followed

### Testing

* \[ ] This PR added unit tests
* \[ ] This PR added integration/end-to-end tests
* \[ ] These changes required manual testing that is documented below
* \[ ] Anything not tested is documented

The following manual testing was done:

* TBD

The following was not tested:

* TBD

<details>
<summary>
Commit message guidelines
</summary>
We use 'Conventional Commits' to ensure that our commit messages are easy to read, follow a consistent format, and for automated release note generation. Please follow the guidelines below when writing your commit messages:

1. BREAKING CHANGE: a commit that has a footer BREAKING CHANGE:, or appends a ! after the type/scope, introduces a breaking API change (correlating with MAJOR in Semantic Versioning). A BREAKING CHANGE can be part of commits of any type.  NOTE: currently breaking changes will only bump the MAJOR version.
2. The title is prefixed with one of the following:

| Prefix    | Description                                         | Semantic Version Update | Captured in Release Notes |
|-----------|-----------------------------------------------------|-------------------------|---------------------------|
| feat:     | a new feature                                       | MINOR                   | Yes                       |
| fix:      | a bug fix                                           | PATCH                   | Yes                       |
| perf:     | performance                                         | PATCH                   | Yes                       |
| refactor: | code change that isn't feature or fix               | none                    | No                        |
| test:     | adding missing tests                                | none                    | No                        |
| docs:     | changes to documentation                            | none                    | Yes                        |
| build:    | changes to build process                            | none                    | No                        |
| ci:       | changes to CI configuration                         | none                    | No                        |
| style:    | formatting, missing semi-colons, etc                | none                    | No                        |
| chore:    | updating grunt tasks etc; no production code change | none                    | No                        |

</details>

### CODE REVIEW

File: src/commands/one-shot/default-one-shot.ts [hunk 1]
```diff
@@ -857,6 +950,9 @@ export class DefaultOneShotCommand extends BaseCommand implements OneShotCommand
                 return argvPushGlobalFlags(argv);
               },
               this.taskList,
+              (): boolean => {
+                return hasExplorers;
```
Comment (jeromy-cannon) [inline src/commands/one-shot/default-one-shot.ts]:
```suggestion
                return !hasExplorers;
```
skip if not has explorers

File: src/commands/one-shot/default-one-shot.ts [hunk 2]
```diff
@@ -876,12 +972,15 @@ export class DefaultOneShotCommand extends BaseCommand implements OneShotCommand
                 return argvPushGlobalFlags(argv);
               },
               this.taskList,
+              (): boolean => {
+                return hasRelays;
```
[the remark this answers is not in the document]
Comment (nathanklick) [reply src/commands/one-shot/default-one-shot.ts]: fixed

File: src/commands/one-shot/default-one-shot.ts (hunk 1 above)
Comment (nathanklick) [reply src/commands/one-shot/default-one-shot.ts]: fixed
